index.src.html

```diff
@@ -2049,111 +2067,131 @@
-    to be "opaque" to avoid early optimization. We should make conversion
-    explicit and user controlled by offering a `videoFrame.convertTo(format)`
-    that returns a Promise containing a new VideoFrame for which the
-    copies/translations are performed.
+### Constructors ###{#videoframe-constructors}
 
 <dfn constructor for=VideoFrame title="VideoFrame(imageBitmap, frameInit)">
   VideoFrame(imageBitmap, frameInit)
 </dfn>
-1. If |frameInit| is not a [=valid VideoFrameInit=], throw a {{TypeError}}.
-2. If the value of |imageBitmap|'s' {{PlatformObject/[[Detached]]}} internal
-    slot is set to `true`, then throw an {{InvalidStateError}} DOMException.
-3. Let |frame| be a new {{VideoFrame}}.
-4. Assign `false` to |frame|’s {{VideoFrame/[[detached]]}} internal slot.
-5. Use a copy of the pixel data in |imageBitmap| to initialize to following
-    frame attributes:
-    1. Initialize `frame.pixelFormat` be the underlying format of imageBitmap.
-    2. Initialize `frame.planes` to describe the arrangement of memory of the
-        copied pixel data.
-    3. Assign regions of the copied pixel data to the
-        {{Plane/[[plane buffer]]}} internal slot of each plane as
-        appropriate for the pixel format.
-    4. Initialize `frame.codedWidth` and `frame.codedHeight` describe the width
-        and height of the imageBitamp prior to any cropping or aspect ratio
-        adjustments.
-6. Use |frameInit| to initialize the remaining frame attributes:
-    1. If `frameInit.cropLeft` is present, initialize it `frame.cropLeft`.
-        Otherwise, default `frame.cropLeft` to zero.
-    2. If `frameInit.cropTop` is present, initialize it to `frame.cropTop`.
-        Otherwise, default `frame.cropTop` to zero.
-    3. If `frameInit.cropWidth` is present, initialize it to `frame.cropWidth`.
-        Otherwise, default `frame.cropWidth` to `frame.codedWidth`.
-    4. If `frameInit.cropHeight` is present, initialize it to
-        `frame.cropHeight`. Otherwise, default `frame.cropHeight` to
-        `frame.codedHeight`.
-    5. If `frameInit.displayWidth` is present, initialize it to
-        `frame.displayWidth`. Otherwise, default `frame.displayWidth` to
-        `frame.codedWidth`.
-    6. If `frameInit.displayHeight` is present, initialize it to
-        `frame.displayHeight`. Otherwise, default `frame.displayHeight` to
-        `frame.codedHeight`.
-    7. If `frameInit.duration` is present, initialize it to `frame.duration`.
-        Otherwise, default `frame.duration` to null.
-    8. If `frameInit.timestamp` is present, initialize it to `frame.timestamp`.
-        Otherwise default `frame.timestamp` to null.
+1. If the value of |imageBitmap|'s' {{PlatformObject/[[Detached]]}} internal
+    slot is `true`, then throw an {{InvalidStateError}}
+    {{DOMException}}.
+2. Let |resource| be the [=frame resource=] containing the pixel data for
+    |imageBitmap|.
+3. Let |resourceReference| be a reference to |resource|.
```

Comment:
How does it work if this data is then later made unavailable, for example because `transferFromImageBitmap` has been called ?

Reply:
@sandersdan friendly nudge to check me on the above.

Reply:
I am not very familiar with the spec details at the level of resource references. `ImageBitmaps` are immutable, so it's always a valid to implement "copy" by reference, and this is compatible with `transferFromImageBitmap()` as long as exclusive access isn't needed (it's never needed, at least in Chrome's implementation).

It's probably fine to just say that we copy the bitmap data, the same as `createImageBitmap(ImageBitmap)` does.

Reply:
I've just re-read a bunch of stuff about this, and I agree with the last sentence in Dan's message above.

Reply:
We're all in agreement. Our understanding is that this doesn't require any change to the language / interfaces (implied behavior).